src/Cookie/CookieJar.php

```diff
@@ -86,6 +86,25 @@ public static function shouldPersist(
         return false;
     }
 
+    /**
+     * Finds and returns the cookie based on the name
+     *
+     * @param string $name cookie name to search for
+     * @return SetCookie|null cookie that was found or null if not found
+     */
+    public function getCookieByName($name)
```

Comment:
Not sure, setCookie has a completely different behaviour than the name suggests, so it might be confusing.